cmd/obfuscate-archive/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/openshift/insights-operator/pkg/anonymization"
+	"github.com/openshift/insights-operator/pkg/gather"
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/recorder"
+	"github.com/openshift/insights-operator/pkg/recorder/diskrecorder"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		_, _ = fmt.Fprintf(os.Stderr, "Path to the archive was not provided\n\n"+
+			"Usage: go run ./cmd/obfuscate-archive/main.go PATH_TO_THE_ARCHIVE\n\n"+
+			"Obfuscates the archive located at PATH_TO_THE_ARCHIVE\n")
+		return
+	}
+
+	path := os.Args[1]
+
+	if err := obfuscateArchive(path); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Unable to obfuscate archive: %v\n", err)
+	}
+}
+
+func obfuscateArchive(path string) error {
+	const suffix = ".tar.gz"
+	if !strings.HasSuffix(path, suffix) {
+		return fmt.Errorf(`invalid path to the archive: should end with "%v"`, suffix)
+	}
+
+	newPath := strings.TrimSuffix(path, suffix) + "-obfuscated" + suffix
+
+	records, err := readArchive(path)
+	if err != nil {
+		return err
+	}
+
+	clusterBaseDomain, err := getClusterBaseDomain(records)
+	if err != nil {
+		return err
+	}
+
+	networks, err := anonymization.GetNetworksForAnonymizerFromRecords(records)
+	if err != nil {
+		return err
+	}
+
+	anonymizer, err := anonymization.NewAnonymizer(clusterBaseDomain, networks, nil)
+	if err != nil {
+		return err
+	}
+
+	var anonymizedRecords record.MemoryRecords
+
+	for _, r := range records {
+		if r.Name == recorder.MetadataRecordName+".json" {
+			var metadata gather.ArchiveMetadata
+
+			err = json.Unmarshal(r.Data, &metadata)
+			if err != nil {
+				return err
+			}
+
+			metadata.IsGlobalObfuscationEnabled = true
+
+			metadataBytes, err := json.Marshal(metadata) //nolint:govet
+			if err != nil {
+				return err
+			}
+
+			r.Data = metadataBytes
+		}
+
+		anonymizedRecords = append(anonymizedRecords, *anonymizer.AnonymizeMemoryRecord(r))
+	}
+
+	diskRecorder := diskrecorder.New("")
+
+	_, err = diskRecorder.SaveAtPath(anonymizedRecords, newPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getClusterBaseDomain(records map[string]*record.MemoryRecord) (string, error) {
+	const filePath = "config/infrastructure.json"
+
+	r, found := records[filePath]
+	if !found {
+		return "", fmt.Errorf(
+			"%v record needed to fetch cluster base domain wasn't found", filePath,
+		)
+	}
+
+	var infrastructure configv1.Infrastructure
+
+	err := json.Unmarshal(r.Data, &infrastructure)
+	if err != nil {
+		return "", err
+	}
+
+	domain := infrastructure.Status.EtcdDiscoveryDomain
+	if len(domain) == 0 {
+		return "", fmt.Errorf("status.etcdDiscoveryDomain from %v is empty", filePath)
```

Comment:
I got this error when I was trying it out. I was using cluster-bot cluster archive (with empty `etcdDiscoveryDomain`. Not sure how often it's empty) . It's probably not a good idea to try to read the domain from e.g `apiServerURL` right? I guess it's just detail....

Reply:
It looks like we can read the domain from the new config map.

Reply:
We can try multiple places including the new one